Gems/ROS2Controllers/Code/Source/RobotControl/Controllers/RigidBodyController/RigidBodyTwistControlComponent.cpp

```diff
@@ -72,23 +88,104 @@ namespace ROS2Controllers
         }
         m_bodyHandle = rigidBody->m_bodyHandle;
         m_sceneFinishSimHandler = AzPhysics::SceneEvents::OnSceneSimulationFinishHandler(
-            [this, sceneInterface]([[maybe_unused]] AzPhysics::SceneHandle sceneHandle, float fixedDeltaTime)
+            [this]([[maybe_unused]] AzPhysics::SceneHandle sceneHandle, float fixedDeltaTime)
             {
-                auto* rigidBody = sceneInterface->GetSimulatedBodyFromHandle(sceneHandle, m_bodyHandle);
-                AZ_Assert(sceneInterface, "No body found for previously given handle");
-
-                // Convert local steering to world frame
-                const AZ::Transform robotTransform = rigidBody->GetTransform();
-                const auto linearVelocityGlobal = robotTransform.TransformVector(m_linearVelocityLocal);
-                const auto angularVelocityGlobal = robotTransform.TransformVector(m_angularVelocityLocal);
-                Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequests::SetLinearVelocity, linearVelocityGlobal);
-                Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequests::SetAngularVelocity, angularVelocityGlobal);
+                OnSceneSimulationFinish(sceneHandle, fixedDeltaTime);
             },
             aznumeric_cast<int32_t>(AzPhysics::SceneEvents::PhysicsStartFinishSimulationPriority::Components));
         sceneInterface->RegisterSceneSimulationFinishHandler(defaultSceneHandle, m_sceneFinishSimHandler);
         AZ::TickBus::Handler::BusDisconnect();
     }
 
+    AZ::Vector3 ComputeImpulse(const AZ::Vector3& errorVec, AZStd::array<PidConfiguration, 3>& controllers, float fixedDeltaTime)
```

Comment:
Best to move non-class methods to the top for clarity. This also looks like a generic feature function, potentially to be refactored out of the file (once reused).